src/network/networkd-netdev-tunnel.c

```diff
@@ -214,6 +215,21 @@ static int netdev_vti_fill_message_create(NetDev *netdev, Link *link, sd_netlink
         if (r < 0)
                 return log_netdev_error_errno(netdev, r, "Could not append IFLA_IPTUN_LINK attribute: %m");
 
+        if (t->key)
```

Comment:
please don't use C's downgrade-to-bool feature except when checking for ptrs and actual bools. i.e. given that `t->key` is actually numeric, please change the check to `if (t->key != 0)` or so.